storage-operator/pkg/controller/volume/volume_controller.go

```diff
@@ -713,6 +713,40 @@ func (self *ReconcileVolume) finalizeVolume(
 	return delayedRequeue(nil)
 }
 
+func (self *ReconcileVolume) refreshDeviceName(
+	ctx context.Context,
+	volume *storagev1alpha1.Volume,
+	pv *corev1.PersistentVolume,
+) (reconcile.Result, error) {
+	nodeName := string(volume.Spec.NodeName)
+	reqLogger := log.WithValues(
+		"Volume.Name", volume.Name, "Volume.NodeName", nodeName,
+	)
+
+	if pv.Spec.PersistentVolumeSource.Local == nil {
+		reqLogger.Info("skipping volume: not a local storage")
+		return endReconciliation()
+	}
+	path := pv.Spec.PersistentVolumeSource.Local.Path
+
+	name, err := self.salt.GetDeviceName(ctx, nodeName, volume.Name, path)
+	if err != nil {
+		self.recorder.Event(
+			volume, corev1.EventTypeNormal, "SaltCall",
+			"device path resolution failed",
+		)
+		reqLogger.Error(err, "cannot get device name from Salt response")
+		return delayedRequeue(err)
+	}
+	if volume.Status.DeviceName != name {
+		volume.Status.DeviceName = name
+		reqLogger.Info("update device name", "DeviceName", name)
```

Comment:
I'm not familiar with how structured logging should work with controller-runtime, but given the `WithValues` call above, shouldn't we also namespace `DeviceName`?

```suggestion
		reqLogger.Info("update device name", "Volume.Status.DeviceName", name)
```

Reply:
I could "namespace" it yeah, I'll use `Volume.DeviceName` (I think the fact that this lives inside the Status sub-resource is an implementation detail, the one inspecting the log doesn't need this knowledge).